Git-egylet/pacman.py

```diff
@@ -0,0 +1,242 @@
+            done (bool): tells whether the game is terminated or not
+            info (str): additional information
+        """
+        # set basic status
+        done = False
+        # get the current PacMan position
+        pos_x, pos_y = self.body
+        # get the new position
+        x, y = self._move(pos_x, pos_y, action)
+        # add current state to the body
+        if (x, y) not in self.body:
+            self.body = (x, y)
+        # create observation of the current state of the game
+        obs = self._create_observation()
+        # save observation
+        self.last_state = obs
+        # check for pellets to eat
+        self.score = self._check_pellets(x, y, self.score)
+        # count the steps
+        self.step_counter += 1
+        # check if the game is terminated
+        done = self.is_done(self.max_steps)
+        # additional information
+        info = None
+
+        return obs.flatten(), self.score, done, info
+
+    def is_done(self, max_steps: int):
+        """
+        This function checks if the game has reached the maximum amount of timesteps.
+        :param max_steps: maximum amount of steps during the game
+        :return: boolean value
+        """
+        if self.step_counter > max_steps:
+            return True
+        else:
+            return False
+
+    def create_pacman(self):
+        """
+        This function creates Pacman and sets its starting position.
+        :return: None
+        """
+        self.body = (0, 0)
+
+    def create_pellets(self, numbers: int):
+        """
+        This function creates the pellets for Pacman to eat in the available positions of the map.
+        :param numbers: the number of pellets to create
+        :return: None
+        """
+        for _ in range(numbers):
+            coordinates = tuple(np.random.randint(0, self.world_size, (2,)))
+            while coordinates in self.pellets or coordinates in self.body:
+                coordinates = tuple(np.random.randint(0, self.world_size, (2,)))
+            self.pellets.append(coordinates)
+
+    def _move(self, pos_x: int, pos_y: int, action: int):
+        """
+        This function is responsible for the movement of Pacman.
+        :param pos_x: horizontal coordinate
+        :param pos_y: vertical coordinate
+        :param action: chosen action by the user between 0-3, which refer to directions
+        :return: x, y (int): the new position of Pacman after the move was made.
+        """
+        self.direction = self._change_direction(action)
+        if self.direction == 0:
+            x, y = self._up(pos_x, pos_y)
+        elif self.direction == 1:
+            x, y = self._right(pos_x, pos_y)
+        elif self.direction == 2:
+            x, y = self._down(pos_x, pos_y)
+        elif self.direction == 3:
+            x, y = self._left(pos_x, pos_y)
+        else:
+            raise NotImplementedError
+        return x, y
+
+    def _change_direction(self, action: int):
```

Comment:
Mivel nincs a class instance-ére vonatkozó változó, lehetne staticmethod is.
```suggestion
    @staticmethod
    def _change_direction(action: int):
```

Reply:
Solved